### PR TITLE
use internal RegexURLResolver api due to removal of reverse() method in django-1.10

### PR DESCRIPTION
**Pull request by pjstevns**

### CODE REVIEW

File: wagtail/contrib/wagtailroutablepage/models.py [hunk 1]
```diff
@@ -63,7 +64,12 @@ def reverse_subpage(self, name, args=None, kwargs=None):
         args = args or []
         kwargs = kwargs or {}
 
-        return self.get_resolver().reverse(name, *args, **kwargs)
+        if DJANGO_VERSION <= (1, 9):
```
Comment (timgraham) [inline wagtail/contrib/wagtailroutablepage/models.py]: Should be `< (1, 10)` otherwise 1.9 bug fix releases are excluded.

**Comment by timgraham** [comment]
I have a good reason for removing the `reverse()` method other than it was unused and untested in Django itself. I might as well add it back to avoid causing you this trouble.

**Comment by timgraham** [comment]
There you go: https://github.com/django/django/pull/6809

**Comment by gasman** [comment]
Fix no longer required in Wagtail, as of https://github.com/django/django/pull/6809.

**Pull request closed without merging**